services/supervisor/src/api.rs

```diff
@@ -15,18 +15,100 @@
 use exonum::{
     blockchain::ConsensusConfig,
     crypto::Hash,
-    runtime::rust::{
-        api::{self, ServiceApiBuilder, ServiceApiState},
-        Broadcaster,
+    helpers::Height,
+    runtime::{
+        rust::{
+            api::{self, ServiceApiBuilder, ServiceApiState},
+            Broadcaster,
+        },
+        ArtifactId,
     },
 };
 use failure::Fail;
+use serde_derive::{Deserialize, Serialize};
+use std::{convert::TryFrom, str::FromStr};
 
 use super::{
     schema::Schema, transactions::SupervisorInterface, ConfigProposalWithHash, ConfigPropose,
     ConfigVote, DeployRequest, SupervisorConfig,
 };
 
+/// Query for retrieving information about deploy state.
+/// This is flattened version of `DeployRequest` which can be
+/// encoded via URL query parameters.
+#[derive(Debug, Clone, PartialEq)]
+#[derive(Serialize, Deserialize)]
+pub struct DeployInfoQuery {
+    /// Artifact identifier as string, e.g. `0:exonum-supervisor:0.13.0-rc.2"
+    pub artifact: String,
+    /// Artifact spec bytes as hexadecimal string.
+    pub spec: String,
+    /// Deadline height.
+    pub deadline_height: u64,
+}
+
+impl TryFrom<DeployInfoQuery> for DeployRequest {
+    type Error = api::Error;
+
+    fn try_from(query: DeployInfoQuery) -> Result<Self, Self::Error> {
+        let artifact = ArtifactId::from_str(&query.artifact)
+            .map_err(|err| api::Error::BadRequest(err.to_string()))?;
+        let spec =
+            hex::decode(query.spec).map_err(|err| api::Error::BadRequest(err.to_string()))?;
+        let deadline_height = Height(query.deadline_height);
+
+        let request = Self {
+            artifact,
+            spec,
+            deadline_height,
+        };
+
+        Ok(request)
+    }
+}
+
+impl From<DeployRequest> for DeployInfoQuery {
+    fn from(request: DeployRequest) -> Self {
+        let artifact = request.artifact.to_string();
+        let spec = hex::encode(&request.spec);
+        let deadline_height = request.deadline_height.0;
+
+        Self {
+            artifact,
+            spec,
+            deadline_height,
+        }
+    }
+}
+
+/// State of the deployment performed by `Supervisor`.
+#[derive(Debug, Clone, PartialEq)]
+#[derive(Serialize, Deserialize)]
+pub enum DeployState {
```

Comment:
Nit: AFAIK, we try to uniformly use `#[serde(rename_all = "snake_case")]` for enums.